x-pack/packages/ml/aiops_log_rate_analysis/ecs_fields.ts

```diff
@@ -0,0 +1,242 @@
+
+  // Event fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+  'event.action',
+  'event.agent_id_status',
+  'event.category',
+  'event.code',
+  'event.dataset',
+  'event.kind',
+  'event.module',
+  'event.outcome',
+  'event.provider',
+  'event.reason',
+  'event.reference',
+  'event.timezone',
+  'event.type',
+  'event.url',
+
+  // File fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-file.html
+  'file.attributes',
+  'file.path',
+  'file.type',
+
+  // Host fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-host.html
+  'host.architecture',
+  'host.domain',
+  'host.ip',
+  'host.name',
+
+  // HTTP fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-http.html
+  'http.request.method',
+  'http.request.mime_type',
+  'http.request.referrer',
+  'http.response.mime_type',
+  'http.response.status_code',
+  'http.version',
+
+  // Log fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-log.html
+  'log.file.path',
+  'log.level',
+  'log.logger',
+  'log.origin.file.name',
+  'log.origin.function',
+
+  // Network fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-network.html
+  'network.application',
+  'network.direction',
+  'network.name',
+  'network.protocol',
+  'network.type',
+
+  // Orchestrator fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-orchestrator.html
+  'orchestrator.cluster.name',
+  'orchestrator.cluster.version',
+  'orchestrator.namespace',
+  'orchestrator.organization',
+  'orchestrator.type',
+
+  // Process fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-process.html
+  'process.command_line',
+  'process.env_vars',
+  'process.name',
+
+  // Rule fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-rule.html
+  'rule.author',
+  'rule.category',
+  'rule.description',
+  'rule.name',
+  'rule.ruleset',
+  'rule.version',
+
+  // Server fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-server.html
+  'server.domain',
+  'server.ip', // IP field
+  'server.port',
+
+  // Service fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-service.html
+  'service.environment',
+  'service.name',
+  'service.node.name',
+  'service.node.role', // deprecated
+  'service.node.roles',
+  'service.state',
+  'service.type',
+  'service.version',
+
+  // Source fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-source.html
+  'source.domain',
+  'source.ip', // IP field
+  'source.port',
+
+  // URL fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-url.html
+  'url.domain',
+  'url.full',
+
+  // User fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-user.html
+  'user.name',
+  'user.domain',
+  'user.email',
+  'user.roles',
+
+  // User agent fields
+  // https://www.elastic.co/guide/en/ecs/current/ecs-user_agent.html
+  'user_agent.device.name',
+  'user_agent.name',
+  'user_agent.original',
+  'user_agent.version',
+];
+
+const ecsFieldsMustHave = [
```

Comment:
As discussed, should we simplify this and only check for `ece.version` ?

Reply:
Updated in [0e38e67](https://github.com/elastic/kibana/pull/188913/commits/0e38e67acb55af4c1a5ea2398ee391f3e919cb7c).

Based on the ECS detection and fields automatically selected, I added some text showing up when applicable in the callout that describes the analysis.

<img width="1104" alt="image" src="https://github.com/user-attachments/assets/dabcb033-f1a4-4fa1-a872-6307caaa6263">

If we identify ECS and pick a selection of fields, we'll show a text like `The source documents were identified as ECS compliant. 16 out of 29 fields were preselected for the analysis. Use the "Fields" dropdown to adjust the selection.` If the index is not ECS and there are more than a 100 fields in total available, we just pick the first 100 fields and display `100 out of 290 fields were preselected for the analysis. Use the "Fields" dropdown to adjust the selection.`. If the index is not ECS and has less than 100 fields, all fields will be selected and no message like this will be added to the callout.